helper_apps/MediaFileChunker/fileChunker.js

```diff
@@ -43,18 +48,57 @@ const generateUniqueFolderName = () => {
     return uniqueOutputPath;
 }
 
+async function downloadFile(url, outputPath) {
+    try {
+        // Make an HTTP request for the file
+        const response = await axios.get(url, { responseType: 'stream' });
+
+        // Create a writable file stream to save the file
+        const fileStream = fs.createWriteStream(outputPath);
+
+        // Pipe the response data into the file stream
+        response.data.pipe(fileStream);
+
+        // Wait for the file stream to finish writing
+        await new Promise((resolve, reject) => {
+            fileStream.on('finish', resolve);
+            fileStream.on('error', reject);
+        });
+
+        console.log(`Downloaded file saved to: ${outputPath}`);
+    } catch (error) {
+        console.error(`Error downloading file from ${url}:`, error);
+        throw error;
+    }
+}
+
 async function splitMediaFile(inputPath, chunkDurationInSeconds = 600) {
     try {
+        // Create unique folder
+        const uniqueOutputPath = generateUniqueFolderName();
+        fs.mkdirSync(uniqueOutputPath, { recursive: true });
+
+        // Download the file if it's not a local file
+        const isUrl = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i.test(inputPath);
+        if (isUrl) {
+            inputPath = ensureEncoded(inputPath);
```

Comment:
Just curious - why don't we know whether inputPath is encoded at this point?

Reply:
we know actually but while testing manually i saw that we could get trapped with this one in direct call so added it